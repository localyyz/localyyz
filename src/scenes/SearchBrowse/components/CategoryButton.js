import React from "react";
import {
  View,
  ImageBackground,
  StyleSheet,
  Text,
  TouchableOpacity
} from "react-native";

// custom
import { Colours, Sizes, Styles } from "~/src/constants";

// constants
export const BUTTON_PADDING = 5;
const WIDTH = Sizes.Width / 2 - BUTTON_PADDING - BUTTON_PADDING / 2;
const HEIGHT = 2 * Sizes.Width / 5;
const SMALL_WIDTH = Sizes.Width / 3;
const SMALL_HEIGHT = Sizes.Width / 6;
const SMALL_RIGHT_MARGIN = Sizes.InnerFrame / 4;

export default class CategoryButton extends React.Component {
  render() {
    const imageStyle = StyleSheet.flatten([
      styles.photo,
      this.props.isSmall && styles.small
    ]);

    return (
      <TouchableOpacity activeOpacity={1} onPress={this.props.onPress}>
        <View style={styles.container}>
          <ImageBackground
            source={{ uri: this.props.imageUrl }}
            style={imageStyle}>
            <View style={styles.overlay} />
            <View style={styles.labelBackground}>
              <Text numberOfLines={2} style={[styles.label, styles.labelLarge]}>
                {this.props.title}
              </Text>
            </View>
          </ImageBackground>
          {this.props.isSmall ? (
            <View style={{ width: SMALL_WIDTH }}>
              <Text numberOfLines={1} style={[styles.label, styles.labelSmall]}>
                {this.props.title}
              </Text>
            </View>
          ) : null}
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "space-between"
  },

  overlay: {
    opacity: 0.5,
    backgroundColor: "#000",
    position: "absolute",
    left: 0,
    right: 0,
    top: 0,
    bottom: 0
  },

  photo: {
    width: WIDTH,
    height: HEIGHT,
    backgroundColor: Colours.MenuBackground,
    overflow: "hidden",
    borderWidth: Sizes.Hairline,
    borderColor: Colours.Border
  },

  small: {
    height: SMALL_HEIGHT,
    width: SMALL_WIDTH,
    marginRight: SMALL_RIGHT_MARGIN,
    borderRadius: 10
  },

  label: {
    ...Styles.Text,
    ...Styles.Emphasized,
    textAlign: "center"
  },

  labelSmall: {
    flex: 1,
    fontSize: Sizes.TinyText
  },

  labelLarge: {
    color: "white",
    textAlign: "center"
  },

  labelBackground: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    padding: WIDTH / 6
  }
});
