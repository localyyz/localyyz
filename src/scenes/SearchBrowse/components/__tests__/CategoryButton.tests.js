import React from "react";
import Renderer from "react-test-renderer";

import CategoryButton from "../CategoryButton";

const DEFAULT_PROPS = {
  id: "activewear",
  type: "activewear",
  title: "Activewear",
  values: [],
  imageUrl:
    "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg"
};

describe("CategoryButton", () => {
  it("should render properly", () => {
    let render = Renderer.create(<CategoryButton {...DEFAULT_PROPS} />);
    expect(render.toJSON()).toMatchSnapshot();
  });

  it("should render properly (isSmall)", () => {
    let render = Renderer.create(<CategoryButton isSmall {...DEFAULT_PROPS} />);
    expect(render.toJSON()).toMatchSnapshot();
  });
});
