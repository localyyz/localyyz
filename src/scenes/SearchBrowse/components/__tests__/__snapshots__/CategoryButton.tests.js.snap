// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryButton should render properly (isSmall) 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "height": 187.5,
        "marginRight": 7.5,
        "width": 250,
      },
      Object {
        "height": 68.18181818181819,
        "marginRight": 3.75,
      },
    ]
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 7.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Activewear
    </Text>
  </View>
</View>
`;

exports[`CategoryButton should render properly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000000",
        "height": 187.5,
        "marginRight": 7.5,
        "width": 250,
      },
      undefined,
    ]
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "https://cdn.shopify.com/s/files/1/1825/3189/products/1589d1c85bfdd70a943bbcc2c75374d6.jpg",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "flex": 1,
        "justifyContent": "space-between",
        "padding": 7.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "borderBottomColor": "#FFFFFF",
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Activewear
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/icons/categories/activewear.png",
        }
      }
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
  </View>
</View>
`;
